eboutic/templates/eboutic/eboutic_main.jinja

```diff
@@ -1,80 +1,190 @@
+          Therefore it is easier and more straightfoward to leave this here #}
+        document.addEventListener('alpine:init', () => {
+            Alpine.data('basket', () => ({
+                items: [{% for item in basket.items.all() %}{
+                        id: {{ item.product_id }},
+                        name: "{{ item.product_name }}",
+                        quantity: {{ item.quantity }},
+                        unit_price: {{ item.product_unit_price }}
+                        },{% endfor %}],
+                total: {{ basket.get_total() }},
+
+
+                add(item) {
+                    item.quantity++;
+                    this.total += item.unit_price;
+                    fetch_add(item.id)
+                        .then(json => this.reload_page_if_diff(json));
+                },
+
 
+                remove(item) {
+                    item.quantity--;
+                    this.total -= item.unit_price;
+                    fetch_remove(item.id)
+                        .then(json => this.reload_page_if_diff(json))
+                },
 
+
+                reload_page_if_diff(json) {
+                    // Reload if json is not well formatted
+                    if (isNaN(parseFloat(json["total"]))) {
+                        window.location.reload();
+                    }
+                    if (Math.abs(json["total"] - this.total) > 0.001) {
+                        // Numbers aren't directly compared because of floating point precision
+                        window.location.reload();
+                    }
+
+                    if (json["items"] === undefined) {
+                        window.location.reload();
+                    }
+                    for (const item of json["items"]) {
+                        if (isNaN(parseInt(item["product_id"])) || isNaN(parseInt(item["quantity"]))) {
+                            window.location.reload();
+                        }
+                        const local_item = this.items.find(i => i.id === item["product_id"])
+                        if (local_item === undefined || local_item.quantity !== item["quantity"]) {
+                            window.location.reload();
+                        }
+                    }
+                },
+
+
+                create_item(id, name, price) {
+                    let new_item = {
+                        id: id,
+                        name: name,
+                        quantity: 0,
+                        unit_price: price
+                    };
+                    this.items.push(new_item);
+                    this.add(new_item);
+                },
+
+
+                add_from_catalog(id, name, price) {
+                    for (let item of this.items) {
+                        if (item.id === id) {
+                            this.add(item);
+                            return;
+                        }
+                    }
```

Comment:
Alors non les for avec les return non non et enfin non. tu fais une boucle conditionnelle ou tu regardes si tu peux utiliser find [MDN](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/find).

Et enfin tu fais un **if** **ET** un **else**

Reply:
Il n'y a objectivement pas de mal à faire ça, c'est clair, lisible, pas trop confus
Moi j'aime pas les else de toute façon, le go m'a corrompu

Reply:
Comme Sli, le return dans un for ne me choque pas. C'est une construction algorithmique qui me semble tout à fait valable. Cependant, je reconnais qu'il est vrai qu'un find aurait été plus concis et plus idiomatique.